pkg/catalog/manager/catalog_sync.go

```diff
@@ -43,6 +47,45 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) (runtime.Object, error) {
 	}
 	logrus.Debugf("Chart hash comparison for global catalog %v: new -- %v --- current -- %v", catalog.Name, commit, catalog.Status.Commit)
 
+	// ensure the system catalog image cache exists
+	var systemCatalogImageCache *v1.ConfigMap
+	if catalog.Name == utils.SystemLibraryName {
+		systemCatalogImageCacheName := utils.GetCatalogImageCacheName(catalog.Name)
+		systemCatalogImageCache, err = m.ConfigMapLister.Get(namespace.System, systemCatalogImageCacheName)
+
+		// if the cache does not exist generate it
+		if err != nil && errors.IsNotFound(err) {
+			systemCatalogImageCache = &v1.ConfigMap{}
+			systemCatalogImageCache.Name = systemCatalogImageCacheName
+			systemCatalogImageCache.Namespace = namespace.System
+			err = image.CreateCatalogImageListConfigMap(systemCatalogImageCache, catalog)
+			if err != nil {
+				return nil, err
+			}
+
+			systemCatalogImageCache, err = m.ConfigMap.Create(systemCatalogImageCache)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+
+		// if the cache exists and is out of date update it
+		if !isUpToDate(commit, catalog) {
+			err = image.CreateCatalogImageListConfigMap(systemCatalogImageCache, catalog)
+			if err != nil {
+				return nil, err
+			}
+
+			// update
+			_, err = m.ConfigMap.Update(systemCatalogImageCache)
```

Comment:
hmmm wouldn't we want to do this after the `traverseAndUpdate` is actually updated? would there be issues if you update the list first but then actual catalog refresh errors out?